fdbserver/Resolver.actor.cpp

```diff
@@ -482,7 +482,7 @@ ACTOR Future<Void> resolveBatch(Reference<Resolver> self,
 				reply.tpcvMap.clear();
 			} else {
 				std::set<uint16_t> writtenTLogs;
-				if (shardChanged || reply.privateMutationCount) {
+				if (shardChanged || reply.privateMutations.size()) {
```

Comment:
`reply.privateMutations` is from `toCommit->getAllMessages();`, which creates one entry for each TLog, even if some TLogs have empty messages. So this will always be true. We need to look inside each entries of `reply.privateMutations` to find out if any really has mutations.

Reply:
See `LogPushData::setMutations()`, there is a header for each TLog's message.